internal/err_helper.go

```diff
@@ -12,3 +14,25 @@ func CloseAndLogError(closer io.Closer, location string) {
 		log.Warnf("unable to close file for location=%q: %+v", location, err)
 	}
 }
+
+type ErrObserve struct {
```

Comment:
nit: I feel that there may be a more descriptive name for this error. Maybe `ErrPath`?

Reply:
Done